modules/java-modules/Makefile.am

```diff
@@ -1,22 +1,26 @@
 if ENABLE_JAVA_MODULES
 
-export GRADLE_OPTS
 
 JAVA_MOD_DST_DIR=$(DESTDIR)/$(moduledir)/java-modules
 MOD_JARS=$(shell find $(abs_top_builddir)/modules/java-modules -name '*.jar' -not -path "$(abs_top_builddir)/modules/java-modules/.gradle/*")
 JAVA_MODULES_INSTALLED_JARS=$(shell find $(JAVA_MOD_DST_DIR) -name '*.jar')
-GRADLE_WORKDIR=$(abs_top_builddir)/modules/java-modules/.gradle
+GRADLE_USER_HOME ?= $(abs_top_builddir)/modules/java-modules/.gradle
+GRADLE_FLAGS ?= --build-cache --project-cache-dir $(GRADLE_USER_HOME)
```

Comment:
Adding changes requested because `--build-cache` option breaks our installer generation.

The build on centos6 uses @czanik 's repository for installing gradle.
https://copr-be.cloud.fedoraproject.org/results/czanik/syslog-ng316epel6/epel-6-x86_64/
(Note: syslog-ng 3.16).

That installs gradle 2.6. With that version.
```
[root@d66dc9ef07a5 yum.repos.d]# /opt/gradle/bin/gradle --build-cache

Unknown command-line option '--build-cache'.
[...]
```

One option to be is to upgrade gradle in our build system.

However, there might be other distributions that do not ship recent enough gradle to support this option. Which means we potentially break building from source for other users too.

Reply:
I have now pushed the setting of --build-cache into the dbld/rules file, which ensures that a recent enough gradle is present (right now 4.10).

At the same time I've attempted to clean up the variable mess in that Makefile.am and although I still think it could be improved by a better integration with gradle, I am not proficient enough in gradle matters to do that.

In any case that latest iteration should fix the behavior with old gradle versions.